src/validators/validation_state.rs

```diff
@@ -47,6 +65,31 @@ impl<'a> ValidationState<'a> {
     pub fn strict_or(&self, default: bool) -> bool {
         self.extra.strict.unwrap_or(default)
     }
+
+    /// Sets the exactness of this state to unknown.
+    ///
+    /// In general this de-optimizes union validation by forcing strict & lax validation passes,
+    /// so it's better to determine exactness and call `set_exactness_ceiling` when possible.
+    pub fn set_exactness_unknown(&mut self) {
+        self.exactness = None;
+    }
+
+    /// Sets the exactness to the lower of the current exactness
+    /// and the given exactness.
+    ///
+    /// This is designed to be used in union validation, where the
+    /// idea is that the "most exact" validation wins.
+    pub fn set_exactness_ceiling(&mut self, exactness: Exactness) {
```

Comment:
shouldn't this been `set_exactness_floor`?

Reply:
or `laxness_ceiling`?

Reply:
I think I might just call this `report_exactness`, given that worked in the PR description.